src/custom-element.js

```diff
@@ -103,8 +103,12 @@ function isTemplateTagSupported() {
  * @return {!Function} The custom element class.
  */
 export function createCustomElementClass(win, name) {
-  const baseCustomElement = createBaseCustomElementClass(win);
-  /** @extends {HTMLElement} */
+  const baseCustomElement = /** @type {Function} */ (
```

Comment:
Function is usually not allowed, this should probably be something like `function(new:WhateverTheTypeIs)`

Reply:
Here's the error I was trying to fix:
```js
src/custom-element.js:111: ERROR - Bad type annotation. Unknown type baseCustomElement
  class CustomAmpElement extends baseCustomElement {
                                 ^
```
I used `{Function}` because it's used as a return type for `createCustomElementClass` and `createBaseCustomElementClass`. If `Function` needs to be changed to `function(new:WhateverTheTypeIs)`, it probably needs to be done in a lot more places. Okay if we punt that to a separate PR? (I want to avoid roping in too many other changes in this already huge PR.)

https://github.com/ampproject/amphtml/blob/4a9ec7291763d35789b6e31a40b6d3b2c13ccb80/src/custom-element.js#L98-L105

https://github.com/ampproject/amphtml/blob/4a9ec7291763d35789b6e31a40b6d3b2c13ccb80/src/custom-element.js#L129-L135